src/plugin.js

```diff
@@ -7,17 +7,13 @@
  * of a Microcosm.
  */
 
-function PluginFactory (config, options, app) {
-  return Object.assign({ app, options }, config)
-}
-
 function checkPlugin (plugin) {
-  if (process.env.NODE_ENV === 'development' && 'register' in plugin && typeof plugin.register !== 'function') {
-    throw TypeError('Expected register property of plugin to be a function, instead got ' + plugin.register)
+  if (process.env.NODE_ENV !== 'production' && typeof plugin !== 'function' && ('register' in plugin && typeof plugin.register !== 'function')) {
+    throw TypeError('Expected plugin to be a function or entity with a method property.')
```

Comment:
this `method property`isn't super clear. You just need a function defined as the property `register`, right?